### PR TITLE
Merge 'main-vs-deps' into UnsignedRightShift

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Need a sign-off for an integration PR

**Comment by AlekseyTs** [comment]
https://github.com/orgs/dotnet/teams/roslyn-compiler Need a sign-off for an integration PR to unblock feature work.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Need a sign-off for an integration PR to unblock feature work.

**Review by jcouv — APPROVED** [review]

**Pull request merged**
